src/common/Notifications.ts

```diff
@@ -1,13 +1,29 @@
 import { AxiosInstance } from 'axios';
 import { Platform, PermissionsAndroid } from 'react-native';
-import {
-  Notifications,
-  Registered,
-  Notification,
-  NotificationCompletion,
-  RegistrationError,
-  NotificationBackgroundFetchResult,
-} from 'react-native-notifications';
+import { useDeveloperConfig } from '../hooks';
+
+// React native notifications automatically initializes Firebase which
+// causes Android app crashes if the user has not registered their app
+// with Firebase services. This function adds safety to not import
+// the dependency unintentionally and still allows all
+// react-native-notifications code to exist.
+export const safelyImportReactNativeNotifications = async () => {
+  const { pushNotificationsConfig } = useDeveloperConfig();
+  // we cant cast to the actual NotificationsRoot at this time due to dynamic imported dependency.
+  // this is true for types used below as well.
+  let rnnotifications: any;
+  if (pushNotificationsConfig?.enabled) {
+    const modulePath = 'react-native-notifications';
+    if (pushNotificationsConfig.applicationName !== 'example') {
+      const resolvePath = require.resolve(modulePath);
```

Comment:
this logic is to prevent app crashes in the r-n-starter app

Reply:
there still exists an issue here - for the Starter App to not crash we need to use an import like:
```
const resolvePath = require.resolve(modulePath);
rnnotifications = await import(resolvePath);
```
for the SDK example app not to crash, we need to use a standard import 
```
rnnotifications = await import(modulePath);
```
the attempted setup is to dynamically import these in a way that is fit for whatever app is being used but the paths are being resolved at compile time (not runtime) so still throwing errors.

im not sure how we can satisfy both at once. ive tried everything i can think of